c_glib/arrow-cuda-glib/cuda.cpp

```diff
@@ -288,7 +288,9 @@ garrow_cuda_buffer_new(GArrowCUDAContext *context,
   auto arrow_context = garrow_cuda_context_get_raw(context);
   auto arrow_buffer = arrow_context->Allocate(size);
   if (garrow::check(error, arrow_buffer, "[cuda][buffer][new]")) {
-    return garrow_cuda_buffer_new_raw(&(*arrow_buffer));
+    std::shared_ptr<arrow::cuda::CudaBuffer> cuda_buffer =
+      arrow_buffer.MoveValueUnsafe();
+    return garrow_cuda_buffer_new_raw(&cuda_buffer);
```

Comment:
Could you use `arrow_` prefix for C++ objects?

```c++
auto arrow_buffer_result = arrow_context->Allocate(size);
if (garrow::check(error, arrow_buffer_result, "[cuda][buffer][new]")) {
  std::shared_ptr<arrow::cuda::CudaBuffer> arrow_buffer = *arrow_buffer_result;
  return garrow_cuda_buffer_new_raw(&arrow_cuda_buffer);
```

Reply:
Ah, ok, I've updated it. (There still needs to be a std::move to convert from unique_ptr to shared_ptr, though.)

Reply:
Thanks!